File: PYTHON/Program/Motorcycle.py
# Saya Rifa Sania NIM 2206697 mengerjakan Latihan Praktikum 4 dalam mata kuliah Desain Pemrograman Berorientasi Objek
# untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin...

from Vehicle import Vehicle

class Motorcycle(Vehicle):
    __jenisMotor = ""
    __kapasitasTangki = ""

    def __init__(self):
        self.__jenisMotor = ""
        self.__kapasitasTangki = ""

    def __init__(self, platNomor="", merk="", tahunProduksi="", warna="", jenisMotor="", kapasitasTangki=""):
        super().__init__(platNomor, merk, tahunProduksi, warna)
        self.__jenisMotor = jenisMotor
        self.__kapasitasTangki = kapasitasTangki

    def setJenisMotor(self, jenisMotor):
        self.__jenisMotor = jenisMotor

    def getJenisMotor(self):
        return self.__jenisMotor    

    def setKapasitasTangki(self, kapasitasTangki):
        self.__kapasitasTangki = kapasitasTangki   

    def getKapasitasTangki(self):
        return self.__kapasitasTangki

File: CPP/Program/Motorcycle.cpp
#pragma once
#include <iostream>
#include <string>
#include "Vehicle.cpp"

using namespace std;

class Motorcycle : public Vehicle{
    private:
        string jenisMotor;
        string kapasitasTangki;

    public:
        Motorcycle(){
        }

        Motorcycle(string platNomor, string merk, string tahunProduksi, string warna, string jenisMotor) : Vehicle(platNomor, merk, tahunProduksi, warna){
            this->jenisMotor = jenisMotor;
            kapasitasTangki = "";
        }

        Motorcycle(string platNomor, string merk, string tahunProduksi, string warna, string jenisMotor, string kapasitasTangki) : Vehicle(platNomor, merk, tahunProduksi, warna){
            this->jenisMotor = jenisMotor;
            this->kapasitasTangki = kapasitasTangki;
        }

        void setJenisMotor(string jenisMoto){
            this->jenisMotor = jenisMotor;
        }

        string getJenisMotor(){
            return jenisMotor;
        }

        void setKapasitasTangki(string kapasitasTangki){
            this->kapasitasTangki = kapasitasTangki;
        }

        string getKapasitasTangki(){
            return kapasitasTangki;
        }

        ~Motorcycle(){
        }
};